akka-stream/src/main/scala/akka/stream/KillSwitch.scala

```diff
@@ -49,13 +49,13 @@ object KillSwitches {
   def singleBidi[T1, T2]: Graph[BidiShape[T1, T1, T2, T2], UniqueKillSwitch] =
     UniqueBidiKillSwitchStage.asInstanceOf[Graph[BidiShape[T1, T1, T2, T2], UniqueKillSwitch]]
 
-  abstract class KillableGraphStageLogic(val terminationSignal: Future[Done], _shape: Shape) extends GraphStageLogic(_shape) {
+  abstract class KillableGraphStageLogic(val shutdownListener: TerminationSignal#Listener, _shape: Shape) extends GraphStageLogic(_shape) {
```

Comment:
`KillSwitches.KillableGraphStageLogic` looks like public API. If that is the case, this would be a big breaking change.

One way to avoid exposing `TerminationSignal` as public API would be to enclose over a `shutdownListener` created in outer scope. We would still have to remove `val terminationSignal: Future[Done]` from the constructor signature though.

Thoughts?